src/auto/list_store.rs

```diff
@@ -0,0 +1,130 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use ListModel;
+use ffi;
+use glib;
+#[cfg(any(feature = "v2_44", feature = "dox"))]
+use glib::object::Downcast;
+use glib::object::IsA;
+#[cfg(any(feature = "v2_44", feature = "dox"))]
+use glib::signal::SignalHandlerId;
+#[cfg(any(feature = "v2_44", feature = "dox"))]
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+#[cfg(any(feature = "v2_44", feature = "dox"))]
+use std::boxed::Box as Box_;
+use std::mem;
+#[cfg(any(feature = "v2_44", feature = "dox"))]
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct ListStore(Object<ffi::GListStore, ffi::GListStoreClass>): ListModel;
+
+    match fn {
+        get_type => || ffi::g_list_store_get_type(),
+    }
+}
+
+impl ListStore {
+    #[cfg(any(feature = "v2_44", feature = "dox"))]
+    pub fn new(item_type: glib::types::Type) -> ListStore {
+        unsafe {
+            from_glib_full(ffi::g_list_store_new(item_type.to_glib()))
+        }
+    }
+}
+
+pub trait ListStoreExt {
+    #[cfg(any(feature = "v2_44", feature = "dox"))]
+    fn append<P: IsA<glib::Object>>(&self, item: &P);
+
+    #[cfg(any(feature = "v2_44", feature = "dox"))]
+    fn insert<P: IsA<glib::Object>>(&self, position: u32, item: &P);
+
+    //#[cfg(any(feature = "v2_44", feature = "dox"))]
+    //fn insert_sorted<P: IsA<glib::Object>, Q: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(&self, item: &P, compare_func: /*Unknown conversion*//*Unimplemented*/CompareDataFunc, user_data: Q) -> u32;
```

Comment:
For these scoped function pointers we should probably add some support to gir also, it's well-defined how they work